<a href="https://colab.research.google.com/github/Nikitatamotia/Winter-Project-/blob/main/Project_activity_using_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In simple terms, trying to understand the given data much better, so that we can make some sense out of it.
Link: https://github.com/Nikitatamotia/Winter-Project-/blob/main/WHD.xls
Used 5 years Dataset from Kaggle and made consolidated excel


In [173]:
import pandas as pd
from google.colab import files 

def combine_whole():
  # added code to combine the 5 dataframes, filling missing values with NA
  url ="https://github.com/Nikitatamotia/Winter-Project-/blob/main/WHD.xls?raw=true"
  Happiness_2015_df = pd.read_excel(url,sheet_name = "2015")
  Happiness_2016_df = pd.read_excel(url,sheet_name = "2016")
  Happiness_2017_df = pd.read_excel(url,sheet_name = "2017")
  Happiness_2018_df = pd.read_excel(url,sheet_name = "2018")
  Happiness_2019_df = pd.read_excel(url,sheet_name = "2019")
  
  Happiness_2015_df["year"]=2015
  Happiness_2016_df["year"]=2016
  Happiness_2017_df["year"]=2017
  Happiness_2018_df["year"]=2018
  Happiness_2019_df["year"]=2019
  
  combine_whole = pd.concat([Happiness_2015_df,Happiness_2016_df,Happiness_2017_df, Happiness_2018_df, Happiness_2019_df],ignore_index = True)
  display (combine_whole)
  combine_whole.to_excel("Combine_happiness.xlsx")
  files.download("Combine_happiness.xlsx")
  return combine_whole
  
# created new dataframe from function above

combine_df = combine_whole()

#test code to check new dataframe contains null values 
actual= combine_df.isnull().values.any()
expected = True 

if actual == expected:
  print("Test passed", actual)
else:
  print("Test failed, expected", expected, "got", actual)




Country  Happiness Rank  Happiness Score  \
0                 Switzerland               1            7.587   
1                     Iceland               2            7.561   
2                     Denmark               3            7.527   
3                      Norway               4            7.522   
4                      Canada               5            7.427   
..                        ...             ...              ...   
777                    Rwanda             152            3.334   
778                  Tanzania             153            3.231   
779               Afghanistan             154            3.203   
780  Central African Republic             155            3.083   
781               South Sudan             156            2.853   

     Economy (GDP per Capita)  Health (Life Expectancy)  Freedom  \
0                     1.39651                   0.94143  0.66557   
1                     1.30232                   0.94784  0.62877   
2                     1.32548                   0.87464  0.64938   
3                     1.45900                   0.88521  0.66973   
4                     1.32629                   0.90563  0.63297   
..                        ...                       ...      ...   
777                   0.35900                   0.61400  0.55500   
778                   0.47600                   0.49900  0.41700   
779                   0.35000                   0.36100  0.00000   
780                   0.02600                   0.10500  0.22500   
781                   0.30600                   0.29500  0.01000   

     Trust (Government Corruption)  Generosity  year  
0                          0.41978     0.29678  2015  
1                          0.14145     0.43630  2015  
2                          0.48357     0.34139  2015  
3                          0.36503     0.34699  2015  
4                          0.32957     0.45811  2015  
..                             ...         ...   ...  
777                        0.41100     0.21700  2019  
778                        0.14700     0.27600  2019  
779                        0.02500     0.15800  2019  
780                        0.03500     0.23500  2019  
781                        0.09100     0.20200  2019  

[782 rows x 9 columns]

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Test passed True


Finding 5 countries in dataset rank 1-5 in 5 years to visiualise is their any common country having sagnificant factor to keep it rank 1-5. 
Each country’s Happiness Score determines its Happiness Rank – which is its relative position among other countries in that specific year. The first row indicates that Switzerland was ranked the happiest country in 2015 with a happiness score of 7.587. Switzerland was ranked first just before Iceland, which scored 7.561. Denmark was ranked third in 2015, and so on, and from data we can find that Denmark, Iceland and Norway positioned ranked between 1-5 in 5 years and their "Happiness Score" close to each other. Interesting to note that these three countries are from Europe. 



In [65]:
import pandas as pd
from google.colab import files 
def Happiest_countries():
  url ="https://github.com/Nikitatamotia/Winter-Project-/blob/main/WHD.xls?raw=true"
  Happiness_2015_df = pd.read_excel(url,sheet_name = "2015")
  Happiness_2016_df = pd.read_excel(url,sheet_name = "2016")
  Happiness_2017_df = pd.read_excel(url,sheet_name = "2017")
  Happiness_2018_df = pd.read_excel(url,sheet_name = "2018")
  Happiness_2019_df = pd.read_excel(url,sheet_name = "2019")


  Happiness_2015_df["year"]=2015
  Happiness_2016_df["year"]=2016
  Happiness_2017_df["year"]=2017
  Happiness_2018_df["year"]=2018
  Happiness_2019_df["year"]=2019
  
  display (Happiness_2015_df.head())
  display (Happiness_2016_df.head())
  display (Happiness_2017_df.head())
  display (Happiness_2018_df.head())
  display (Happiness_2019_df.head())

  combine_whole = pd.concat([Happiness_2015_df.head(),Happiness_2016_df.head(),Happiness_2017_df.head(),Happiness_2018_df.head(),Happiness_2019_df.head()],ignore_index = True)
  combine_whole.to_excel("seperate_year_data.xlsx")
  files.download("seperate_year_data.xlsx")
  #return combine_whole
Happiest_countries ()

Country  Happiness Rank  Happiness Score  Economy (GDP per Capita)  \
0  Switzerland               1            7.587                   1.39651   
1      Iceland               2            7.561                   1.30232   
2      Denmark               3            7.527                   1.32548   
3       Norway               4            7.522                   1.45900   
4       Canada               5            7.427                   1.32629   

   Health (Life Expectancy)  Freedom  Trust (Government Corruption)  \
0                   0.94143  0.66557                        0.41978   
1                   0.94784  0.62877                        0.14145   
2                   0.87464  0.64938                        0.48357   
3                   0.88521  0.66973                        0.36503   
4                   0.90563  0.63297                        0.32957   

   Generosity  year  
0     0.29678  2015  
1     0.43630  2015  
2     0.34139  2015  
3     0.34699  2015  
4     0.45811  2015

Country  Happiness Rank  Happiness Score  Economy (GDP per Capita)  \
0      Denmark               1            7.526                   1.44178   
1  Switzerland               2            7.509                   1.52733   
2      Iceland               3            7.501                   1.42666   
3       Norway               4            7.498                   1.57744   
4      Finland               5            7.413                   1.40598   

   Health (Life Expectancy)  Freedom  Trust (Government Corruption)  \
0                   0.79504  0.57941                        0.44453   
1                   0.86303  0.58557                        0.41203   
2                   0.86733  0.56624                        0.14975   
3                   0.79579  0.59609                        0.35776   
4                   0.81091  0.57104                        0.41004   

   Generosity  year  
0     0.36171  2016  
1     0.28083  2016  
2     0.47678  2016  
3     0.37895  2016  
4     0.25492  2016

Country  Happiness Rank  Happiness Score  Economy (GDP per Capita)  \
0       Norway               1            7.537                  1.616463   
1      Denmark               2            7.522                  1.482383   
2      Iceland               3            7.504                  1.480633   
3  Switzerland               4            7.494                  1.564980   
4      Finland               5            7.469                  1.443572   

   Health (Life Expectancy)   Freedom  Trust (Government Corruption)  \
0                  0.796667  0.635423                       0.315964   
1                  0.792566  0.626007                       0.400770   
2                  0.833552  0.627163                       0.153527   
3                  0.858131  0.620071                       0.367007   
4                  0.809158  0.617951                       0.382612   

   Generosity  year  
0    0.362012  2017  
1    0.355280  2017  
2    0.475540  2017  
3    0.290549  2017  
4    0.245483  2017

Country  Happiness Rank  Happiness Score  Economy (GDP per Capita)  \
0      Finland               1            7.632                     1.305   
1       Norway               2            7.594                     1.456   
2      Denmark               3            7.555                     1.351   
3      Iceland               4            7.495                     1.343   
4  Switzerland               5            7.487                     1.420   

   Health (Life Expectancy)  Freedom  Trust (Government Corruption)  \
0                     0.874    0.681                          0.393   
1                     0.861    0.686                          0.340   
2                     0.868    0.683                          0.408   
3                     0.914    0.677                          0.138   
4                     0.927    0.660                          0.357   

   Generosity  year  
0       0.202  2018  
1       0.286  2018  
2       0.284  2018  
3       0.353  2018  
4       0.256  2018

Country  Happiness Rank  Happiness Score  Economy (GDP per Capita)  \
0      Finland               1            7.769                     1.340   
1      Denmark               2            7.600                     1.383   
2       Norway               3            7.554                     1.488   
3      Iceland               4            7.494                     1.380   
4  Netherlands               5            7.488                     1.396   

   Health (Life Expectancy)  Freedom  Trust (Government Corruption)  \
0                     0.986    0.596                          0.393   
1                     0.996    0.592                          0.410   
2                     1.028    0.603                          0.341   
3                     1.026    0.591                          0.118   
4                     0.999    0.557                          0.298   

   Generosity  year  
0       0.153  2019  
1       0.252  2019  
2       0.271  2019  
3       0.354  2019  
4       0.322  2019

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Finding 5 countries in dataset rank lowest in 5 years to visiualise is their any common country having sagnificant factor to keep it consistently low in 5 years. 

Dataframe from lowest "Happiness Score" show directly proportional to "Economy (GDP per Capita)" and countries majorly from west africa and Arab. 



In [163]:
import pandas as pd
from google.colab import files 
def Lowest_Happiest_countries():
  
  url ="https://github.com/Nikitatamotia/Winter-Project-/blob/main/WHD.xls?raw=true"
  Happiness_2015_df = pd.read_excel(url,sheet_name = "2015")
  Happiness_2016_df = pd.read_excel(url,sheet_name = "2016")
  Happiness_2017_df = pd.read_excel(url,sheet_name = "2017")
  Happiness_2018_df = pd.read_excel(url,sheet_name = "2018")
  Happiness_2019_df = pd.read_excel(url,sheet_name = "2019")


  Happiness_2015_df["year"]=2015
  Happiness_2016_df["year"]=2016
  Happiness_2017_df["year"]=2017
  Happiness_2018_df["year"]=2018
  Happiness_2019_df["year"]=2019
  
  display (Happiness_2015_df.tail())
  display (Happiness_2016_df.tail())
  display (Happiness_2017_df.tail())
  display (Happiness_2018_df.tail())
  display (Happiness_2019_df.tail())

  combine_whole = pd.concat([Happiness_2015_df.tail(),Happiness_2016_df.tail(),Happiness_2017_df.tail(),Happiness_2018_df.tail(),Happiness_2019_df.tail()],ignore_index = True)
  combine_whole.to_excel("lowest_country_data.xlsx")
  files.download("lowest_country_data.xlsx")
  #return combine_whole
Lowest_Happiest_countries()

Country  Happiness Rank  Happiness Score  Economy (GDP per Capita)  \
153   Rwanda             154            3.465                   0.22208   
154    Benin             155            3.340                   0.28665   
155    Syria             156            3.006                   0.66320   
156  Burundi             157            2.905                   0.01530   
157     Togo             158            2.839                   0.20868   

     Health (Life Expectancy)  Freedom  Trust (Government Corruption)  \
153                   0.42864  0.59201                        0.55191   
154                   0.31910  0.48450                        0.08010   
155                   0.72193  0.15684                        0.18906   
156                   0.22396  0.11850                        0.10062   
157                   0.28443  0.36453                        0.10731   

     Generosity  year  
153     0.22628  2015  
154     0.18260  2015  
155     0.47179  2015  
156     0.19727  2015  
157     0.16681  2015

Country  Happiness Rank  Happiness Score  Economy (GDP per Capita)  \
152        Benin             153            3.484                   0.39499   
153  Afghanistan             154            3.360                   0.38227   
154         Togo             155            3.303                   0.28123   
155        Syria             156            3.069                   0.74719   
156      Burundi             157            2.905                   0.06831   

     Health (Life Expectancy)  Freedom  Trust (Government Corruption)  \
152                   0.21028  0.39747                        0.06681   
153                   0.17344  0.16430                        0.07112   
154                   0.24811  0.34678                        0.11587   
155                   0.62994  0.06912                        0.17233   
156                   0.15747  0.04320                        0.09419   

     Generosity  year  
152     0.20180  2016  
153     0.31268  2016  
154     0.17517  2016  
155     0.48397  2016  
156     0.20290  2016

Country  Happiness Rank  Happiness Score  \
150                    Rwanda             151            3.471   
151                     Syria             152            3.462   
152                  Tanzania             153            3.349   
153                   Burundi             154            2.905   
154  Central African Republic             155            2.693   

     Economy (GDP per Capita)  Health (Life Expectancy)   Freedom  \
150                  0.368746                  0.326425  0.581844   
151                  0.777153                  0.500533  0.081539   
152                  0.511136                  0.364509  0.390018   
153                  0.091623                  0.151611  0.059901   
154                  0.000000                  0.018773  0.270842   

     Trust (Government Corruption)  Generosity  year  
150                       0.455220    0.252756  2017  
151                       0.151347    0.493664  2017  
152                       0.066035    0.354256  2017  
153                       0.084148    0.204435  2017  
154                       0.056565    0.280876  2017

Country  Happiness Rank  Happiness Score  \
151                     Yemen             152            3.355   
152                  Tanzania             153            3.303   
153               South Sudan             154            3.254   
154  Central African Republic             155            3.083   
155                   Burundi             156            2.905   

     Economy (GDP per Capita)  Health (Life Expectancy)  Freedom  \
151                     0.442                     0.343    0.244   
152                     0.455                     0.381    0.481   
153                     0.337                     0.177    0.112   
154                     0.024                     0.010    0.305   
155                     0.091                     0.145    0.065   

     Trust (Government Corruption)  Generosity  year  
151                          0.064       0.083  2018  
152                          0.097       0.270  2018  
153                          0.106       0.224  2018  
154                          0.038       0.218  2018  
155                          0.076       0.149  2018

Country  Happiness Rank  Happiness Score  \
151                    Rwanda             152            3.334   
152                  Tanzania             153            3.231   
153               Afghanistan             154            3.203   
154  Central African Republic             155            3.083   
155               South Sudan             156            2.853   

     Economy (GDP per Capita)  Health (Life Expectancy)  Freedom  \
151                     0.359                     0.614    0.555   
152                     0.476                     0.499    0.417   
153                     0.350                     0.361    0.000   
154                     0.026                     0.105    0.225   
155                     0.306                     0.295    0.010   

     Trust (Government Corruption)  Generosity  year  
151                          0.411       0.217  2019  
152                          0.147       0.276  2019  
153                          0.025       0.158  2019  
154                          0.035       0.235  2019  
155                          0.091       0.202  2019

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

for clarity, using link to access dataframe to get the headers and count of rows and columns.

In [38]:
import pandas as pd
def sorted_happiness():
  # added code to combine the 5 dataframes, filling missing values with NA
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  display (happiness.columns, happiness.shape)
  

happiness=""
sorted_happiness()


Index(['Unnamed: 0', 'Country', 'Happiness Rank', 'Happiness Score',
       'Economy (GDP per Capita)', 'Health (Life Expectancy)', 'Freedom',
       'Trust (Government Corruption)', 'Generosity', 'year'],
      dtype='object')

(782, 10)

Data readed into variable called happiness from the Excel file on Happiness_Data
link: https://github.com/Nikitatamotia/Winter-Project-/blob/main/Happiness_Dataset.xlsx

Used function called sorted_rank to display first 5 rows of data to figured out happiness on scale 

The data is currently sorted by Happiness Rank...

sort the data by Happiness Score in ascending order

displayed sorted table

In [82]:
import pandas as pd
def sorted_rank(df):
# add code below to return a series sorted by Happiness Score in decending order
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx?raw=true"
  happiness = pd.read_excel(url)
  sorted_table = happiness.sort_values(["Happiness Rank"])[["Happiness Score"]].head()
  display(sorted_table)
  return sorted_table

# The code below will run code to see if the first 5 row of series
happiness = "" 
sorted_rank(happiness).index[0]

Happiness Score
0              7.587
470            7.632
158            7.526
625            7.769
315            7.537

0

Checking to see if any feature has empty/missing values, 
output : The 'Trust (Government Corruption)' column has one empty row, this will be replaced by median imputation

In [43]:
import pandas as pd
def sorted_happiness():
  # added code to combine the 5 dataframes, filling missing values with NA
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  display (happiness.isnull().sum())

happiness=""
sorted_happiness()


Unnamed: 0                       0
Country                          0
Happiness Rank                   0
Happiness Score                  0
Economy (GDP per Capita)         0
Health (Life Expectancy)         0
Freedom                          0
Trust (Government Corruption)    1
Generosity                       0
year                             0
dtype: int64


NULL value is in country United Arab Emirates in "Trust (Government Corruption) Column.

In [2]:
import pandas as pd
def sorted_happiness():
  # added code to combine the 5 dataframes, filling missing values with NA
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  NULLS = happiness[happiness.isnull().any(axis=1)]
  display (NULLS.head())

happiness=""
sorted_happiness()

Unnamed: 0               Country  Happiness Rank  Happiness Score  \
489         489  United Arab Emirates              20            6.774   

     Economy (GDP per Capita)  Health (Life Expectancy)  Freedom  \
489                     2.096                      0.67    0.284   

     Trust (Government Corruption)  Generosity  year  
489                            NaN       0.186  2018

Generated new dataframe where missing value is dfropped and saved as Link : https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx

In [81]:
import pandas as pd
from google.colab import files 

def sorted_happiness():
  # added code to combine the 5 dataframes, filling missing values with NA
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  #happiness.dropna(inplace=True)
  happiness2=happiness.copy()
  happiness2.dropna(inplace=True)
  #display (happiness.isnull().sum())
  display(happiness2.isnull().sum())
  
  happiness2.to_excel("Combine_happiness 2.xlsx")
  files.download("Combine_happiness 2.xlsx")


happiness=""
sorted_happiness()

Unnamed: 0                       0
Country                          0
Happiness Rank                   0
Happiness Score                  0
Economy (GDP per Capita)         0
Health (Life Expectancy)         0
Freedom                          0
Trust (Government Corruption)    0
Generosity                       0
year                             0
dtype: int64

checking for duplicate values

Used both the combined dataframes generated before dropped NULL Value and after dropped NULL value, for to check duplicate data and run the code, 

Output : 0,0 from data frames assured there is no duplicate data available. 


In [84]:
import pandas as pd
def sorted_happiness():
  # added code to combine the 5 dataframes, filling missing values with NA
  url_1="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  url_2="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx?raw=true"
  happiness = pd.read_excel(url_1)
  happiness_2 = pd.read_excel(url_2)
  display ( happiness.duplicated().sum())
  display ( happiness_2.duplicated().sum())

happiness=""
happiness2=""
sorted_happiness()

0

0

We see that after cleaning we have 781 rows of clean data with no null values. There are 9 columns and three dtypes(int, float and object)

The data is now clean and void of unnecessary features, we can now proceed to visualizing the data to see the relationship between features

In [29]:
import pandas as pd
def sorted_happiness():
  # added code to combine the 5 dataframes, filling missing values with NA
  #url_1="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  url_2="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx?raw=true"
  #happiness = pd.read_excel(url_1)
  happiness_2 = pd.read_excel(url_2)
  display ( happiness_2.info())

happiness2=""
sorted_happiness()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 781 entries, 0 to 780
Data columns (total 10 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Unnamed: 0                     781 non-null    int64  
 1   Country                        781 non-null    object 
 2   Happiness Rank                 781 non-null    int64  
 3   Happiness Score                781 non-null    float64
 4   Economy (GDP per Capita)       781 non-null    float64
 5   Health (Life Expectancy)       781 non-null    float64
 6   Freedom                        781 non-null    float64
 7   Trust (Government Corruption)  781 non-null    float64
 8   Generosity                     781 non-null    float64
 9   year                           781 non-null    int64  
dtypes: float64(6), int64(3), object(1)
memory usage: 61.1+ KB


None

This is used to view basic statistical details like percentile, mean, std etc

In [ ]:
import pandas as pd
from google.colab import files 
def sorted_calculation():
  # added code to combine the 5 dataframes, filling missing values with NA
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  combine_calculation = happiness.describe()
  display (combine_calculation)
  combine_calculation.to_excel("combine_calculation.xlsx")
  files.download("combine_calculation.xlsx")

happiness=""
sorted_calculation()

Unnamed: 0  Happiness Rank  Happiness Score  Economy (GDP per Capita)  \
count  782.000000      782.000000       782.000000                782.000000   
mean   390.500000       78.698210         5.379018                  0.916047   
std    225.888247       45.182384         1.127456                  0.407340   
min      0.000000        1.000000         2.693000                  0.000000   
25%    195.250000       40.000000         4.509750                  0.606500   
50%    390.500000       79.000000         5.322000                  0.982205   
75%    585.750000      118.000000         6.189500                  1.236187   
max    781.000000      158.000000         7.769000                  2.096000   

       Health (Life Expectancy)     Freedom  Trust (Government Corruption)  \
count                782.000000  782.000000                     781.000000   
mean                   0.612416    0.411091                       0.125436   
std                    0.248309    0.152880                       0.105816   
min                    0.000000    0.000000                       0.000000   
25%                    0.440183    0.309768                       0.054000   
50%                    0.647310    0.431000                       0.091000   
75%                    0.808000    0.531000                       0.156030   
max                    1.141000    0.724000                       0.551910   

       Generosity         year  
count  782.000000   782.000000  
mean     0.218576  2016.993606  
std      0.122321     1.417364  
min      0.000000  2015.000000  
25%      0.130000  2016.000000  
50%      0.201982  2017.000000  
75%      0.278832  2018.000000  
max      0.838075  2019.000000

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

By passing Year as the index parameter, I chose to group our data by country. The output is a pivot table that displays the 2 index, and the 5 different values. I

Link: 
https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx

output: we can already see factors high GDP,Low Government Corruption do major contribution in "Happiness Score" which is higher in Denmark, makes this country in rank 1-5 of five consecutive years. 

Burundi corruption and economy is majorly crisis this country, facing. 

In [32]:
import pandas as pd
from google.colab import files

def sorted_data():

  # added code to combine the 5 dataframes, filling missing values with NA
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx?raw=true"
  happiness = pd.read_excel(url)
  Denmark_happiness=happiness[happiness["Country"]=="Denmark"]
  Burundi_happiness=happiness[happiness["Country"]=="Burundi"]
  Denmark_happiness["Country"]=="Denmark"
  Burundi_happiness["Country"]=="Burundi"
  H_Denmark= pd.pivot_table(Denmark_happiness,index =['Country','year'], values=['Happiness Score','Economy (GDP per Capita)','Trust (Government Corruption)'])
  H_Burundi= pd.pivot_table(Burundi_happiness,index =['Country','year'], values=['Happiness Score','Economy (GDP per Capita)','Trust (Government Corruption)'])
  display (H_Denmark)
  display (H_Burundi)
  combine_whole = pd.concat([Denmark_happiness,Burundi_happiness],ignore_index = True)
  combine_whole.to_excel("highest_lowest_happiness.xlsx")
  files.download("highest_lowest_happiness.xlsx")
  
happiness=""
sorted_data()

Economy (GDP per Capita)  Happiness Score  \
Country year                                              
Denmark 2015                  1.325480            7.527   
        2016                  1.441780            7.526   
        2017                  1.482383            7.522   
        2018                  1.351000            7.555   
        2019                  1.383000            7.600   

              Trust (Government Corruption)  
Country year                                 
Denmark 2015                        0.48357  
        2016                        0.44453  
        2017                        0.40077  
        2018                        0.40800  
        2019                        0.41000

Economy (GDP per Capita)  Happiness Score  \
Country year                                              
Burundi 2015                  0.015300            2.905   
        2016                  0.068310            2.905   
        2017                  0.091623            2.905   
        2018                  0.091000            2.905   
        2019                  0.046000            3.775   

              Trust (Government Corruption)  
Country year                                 
Burundi 2015                       0.100620  
        2016                       0.094190  
        2017                       0.084148  
        2018                       0.076000  
        2019                       0.180000

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Countries have the highest GDP per capita over the years




In [64]:
import pandas as pd
from google.colab import files 
def sort_cal():
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx?raw=true"
  happiness = pd.read_excel(url)
  rich_countries = happiness[['Country','Economy (GDP per Capita)']].groupby('Country').mean().sort_values(by=['Economy (GDP per Capita)'], ascending=False).head()
  display (rich_countries)
  rich_countries.head().to_excel("GDP_cal.xlsx")
  files.download("GDP_cal.xlsx")

happiness=""
sort_cal()

Economy (GDP per Capita)
Country                                          
Qatar                                    1.743691
Luxembourg                               1.637675
Singapore                                1.592138
Kuwait                                   1.555662
Hong Kong S.A.R., China                  1.551675

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

countries have the highest life expectancy over the years?

In [65]:
import pandas as pd
from google.colab import files 
def sort_cal():
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx?raw=true"
  happiness = pd.read_excel(url)
  healthy_countries = happiness[['Country', 'Health (Life Expectancy)']].sort_values(by='Health (Life Expectancy)', ascending=False).head()
  display (healthy_countries)
  healthy_countries.head().to_excel("Health_cal.xlsx")
  files.download("Health_cal.xlsx")

happiness=""
sort_cal()

Country  Health (Life Expectancy)
658    Singapore                     1.141
700    Hong Kong                     1.122
682        Japan                     1.088
654        Spain                     1.062
630  Switzerland                     1.052

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [33]:
import pandas as pd 
def sorted_year(df):
# code below to return a data sorted by highest and lowest Happiness Score of particular country in 2015
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx?raw=true"
  happiness = pd.read_excel(url)
  happiness_Year = happiness.groupby('year')
  sorted_year = happiness_Year['Happiness Score'].describe().sort_values(by="mean",ascending=False).head(10)
  display(sorted_year)
  #return sorted_year


happiness = "" 
sorted_year(happiness)



count      mean       std    min     25%     50%      75%    max
year                                                                  
2019  156.0  5.407096  1.113120  2.853  4.5445  5.3795  6.18450  7.769
2016  157.0  5.382185  1.141674  2.905  4.4040  5.3140  6.26900  7.526
2015  158.0  5.375734  1.145010  2.839  4.5260  5.2325  6.24375  7.587
2018  155.0  5.366897  1.117433  2.905  4.4515  5.3580  6.15400  7.632
2017  155.0  5.354019  1.131230  2.693  4.5055  5.2790  6.10150  7.537

Sorted the data by Freedom and Trust (Government Corruption) in descending order and showed the Country in five different years

In [34]:
import pandas as pd
def sorted_year(df):
# code below to return a series sorted by highest Happiness Score
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness_2019_df = pd.read_excel(url)
  happiness_2019_df["year"] = 2019
  year_2019= happiness_2019_df[happiness_2019_df["year"]==2019].head()
  display (year_2019)

happiness_2019_df = " "
sorted_year(happiness_2019_df)


Unnamed: 0      Country  Happiness Rank  Happiness Score  \
0           0  Switzerland               1            7.587   
1           1      Iceland               2            7.561   
2           2      Denmark               3            7.527   
3           3       Norway               4            7.522   
4           4       Canada               5            7.427   

   Economy (GDP per Capita)  Health (Life Expectancy)  Freedom  \
0                   1.39651                   0.94143  0.66557   
1                   1.30232                   0.94784  0.62877   
2                   1.32548                   0.87464  0.64938   
3                   1.45900                   0.88521  0.66973   
4                   1.32629                   0.90563  0.63297   

   Trust (Government Corruption)  Generosity  year  
0                        0.41978     0.29678  2019  
1                        0.14145     0.43630  2019  
2                        0.48357     0.34139  2019  
3                        0.36503     0.34699  2019  
4                        0.32957     0.45811  2019

In [193]:
import pandas as pd
def sorted_country(df):
# code below to return a series sorted by highest Happiness Score
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  sorted_country =len(happiness[happiness['Happiness Score'] > 7])
  display (sorted_country)

happiness = " "
sorted_country(happiness)

72

In [75]:
import pandas as pd
def country_count():

# code below to return a series sorted by highest Happiness Score
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  HD_2015 = happiness[happiness["year"] == 2015]
  country_count = len(HD_2015[HD_2015['Happiness Score'] > 7])
  display (country_count)
  #return country_count

happiness = " "
country_count()



15

In [76]:
import pandas as pd
def sorted_country(df):

# code below to return a series sorted by highest Happiness Score
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  HD_2016 = happiness[happiness['year'] == 2016]
  sorted_country =len(HD_2016[HD_2016['Happiness Score'] > 7])
  display (sorted_country)

happiness = " "
sorted_country(happiness)

15

In [77]:
import pandas as pd
def sorted_country(df):

# code below to return a series sorted by highest Happiness Score
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  HD_2017 = happiness[happiness['year'] == 2017]
  sorted_country =len(HD_2017[HD_2017['Happiness Score'] > 7])
  display (sorted_country)

happiness = " "
sorted_country(happiness)

13

In [78]:
import pandas as pd
def sorted_country(df):

# code below to return a series sorted by highest Happiness Score
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  HD_2018 = happiness[happiness['year'] == 2018]
  sorted_country =len(HD_2018[HD_2018['Happiness Score'] > 7])
  display (sorted_country)

happiness = " "
sorted_country(happiness)

13

In [79]:
import pandas as pd
def sorted_country(df):

# code below to return a series sorted by highest Happiness Score
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  HD_2019 = happiness[happiness['year'] == 2019]
  sorted_country =len(HD_2019[HD_2019['Happiness Score'] > 7])
  display (sorted_country)

happiness = " "
sorted_country(happiness)

16

In [11]:
import pandas as pd

def sorted_correlation():

# code below to return a series sorted by highest Happiness Score
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  HD_correlation = happiness.corr(method ="pearson", min_periods=5)["Happiness Score"].sort_values(ascending=False)
  display (HD_correlation)
  #return HD_correlation

happiness = " "
sorted_correlation()

Happiness Score                  1.000000
Economy (GDP per Capita)         0.789284
Health (Life Expectancy)         0.742456
Freedom                          0.551258
Trust (Government Corruption)    0.398418
Generosity                       0.137578
year                             0.007065
Unnamed: 0                      -0.191535
Happiness Rank                  -0.992066
Name: Happiness Score, dtype: float64

It is very easy to see a strong positive correlation between the GDP per capita and the score of happiness between countries. In fact there are similar positives correlations like that with our other factor variables. 

Our GDP variable has the strongest correlation with our happiness variable (0.79), after that comes life expectancy (0.74) and Freedom (0.55). The generosity felt within a country variable has the lowest correlation (0.13). 

In [239]:
import pandas as pd

def sorted_correlation():

# code below to return a series sorted by highest Happiness Score
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness.xlsx?raw=true"
  happiness = pd.read_excel(url)
  HD_correlation = happiness.corr(method="pearson", min_periods=5)["Happiness Score"].abs().sort_values(ascending=False)
  display (HD_correlation)
  #return HD_correlation

happiness = " "
sorted_correlation()

Happiness Score                  1.000000
Happiness Rank                   0.992066
Economy (GDP per Capita)         0.789284
Health (Life Expectancy)         0.742456
Freedom                          0.551258
Trust (Government Corruption)    0.398418
Unnamed: 0                       0.191535
Generosity                       0.137578
year                             0.007065
Name: Happiness Score, dtype: float64

The strongest correlation in our correlation matrix (other than the 1.00 for variables with themselves) is between GDP and life expectancy. A larger GDP per capita leads to a larger life expectancy, and vice versa.

In [28]:
import pandas as pd

def sorted_correlation():

# code below to return a series sorted by highest Happiness Score
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx?raw=true"
  happiness = pd.read_excel(url)
  #country_total = happiness.groupby('Country name')['Total_score'].mean().sort_values(ascending= False).head(10)
  #cor = happiness.groupby('Country')['Economy (GDP per Capita)','Health (Life Expectancy)'].mean()
  cor = happiness['Economy (GDP per Capita)'].corr(happiness['Health (Life Expectancy)'])
  T_F_cor = happiness['Trust (Government Corruption)'].corr(happiness['Freedom'])
  F_H_cor = happiness['Freedom'].corr(happiness['Happiness Score'])
  G_H_cor = happiness['Generosity'].corr(happiness['Health (Life Expectancy)'])
  display (cor)
  display(T_F_cor)
  display(F_H_cor)
  display ( G_H_cor)

happiness = " "
sorted_correlation()


0.7877519964961293

0.45959327821751544

0.5533649208776613

0.010718147107628636

using link : https://github.com/Nikitatamotia/Winter-Project-/blob/main/WHD.xls

In [75]:
import pandas as pd
def sorted_GDP(df):
# code below to return a series sorted by highest Happiness Score
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx?raw=true"
  happiness = pd.read_excel(url)
  Highest_GDP = happiness.sort_values(["Happiness Rank","Freedom","Economy (GDP per Capita)"]).head()
  display(Highest_GDP)
  #return Highest_GDP
# The code below will run and test your code to see if the first row of your series is correct 
happiness = "" 
sorted_GDP(happiness)

Unnamed: 0      Country  Happiness Rank  Happiness Score  \
158         158      Denmark               1            7.526   
625         626      Finland               1            7.769   
315         315       Norway               1            7.537   
0             0  Switzerland               1            7.587   
470         470      Finland               1            7.632   

     Economy (GDP per Capita)  Health (Life Expectancy)   Freedom  \
158                  1.441780                  0.795040  0.579410   
625                  1.340000                  0.986000  0.596000   
315                  1.616463                  0.796667  0.635423   
0                    1.396510                  0.941430  0.665570   
470                  1.305000                  0.874000  0.681000   

     Trust (Government Corruption)  Generosity  year  
158                       0.444530    0.361710  2016  
625                       0.393000    0.153000  2019  
315                       0.315964    0.362012  2017  
0                         0.419780    0.296780  2015  
470                       0.393000    0.202000  2018

sorted multiple columns, display the first rows
function called get_gdp_health used which sorted the data by Economy (GDP per Capita) and Health (Life Expectancy) in ascending order
display the first 5 rows of sorted data

In [78]:
import pandas as pd

def get_gdp_health():
  # add code below to return a 5 row series which sorts rows by Economy and Health in decending order 
  url="https://github.com/Nikitatamotia/Winter-Project-/blob/main/Combine_happiness%202.xlsx?raw=true"
  happiness = pd.read_excel(url)
  sorted_data = happiness.sort_values(by = ["Economy (GDP per Capita)", "Health (Life Expectancy)"],ascending=False).head()
  display (sorted_data)
  #return sorted_data

happiness=""
get_gdp_health()





Unnamed: 0     Country  Happiness Rank  Happiness Score  \
349         349       Qatar              35            6.375   
193         193       Qatar              36            6.375   
332         332  Luxembourg              18            6.863   
177         177  Luxembourg              20            6.871   
340         340   Singapore              26            6.572   

     Economy (GDP per Capita)  Health (Life Expectancy)   Freedom  \
349                  1.870766                  0.710098  0.604131   
193                  1.824270                  0.717230  0.566790   
332                  1.741944                  0.845089  0.596628   
177                  1.697520                  0.845420  0.548700   
340                  1.692278                  0.949492  0.549841   

     Trust (Government Corruption)  Generosity  year  
349                       0.439299    0.330474  2017  
193                       0.480490    0.323880  2016  
332                       0.318834    0.283181  2017  
177                       0.353290    0.275710  2016  
340                       0.464308    0.345966  2017